source/plugin/advanced/commands/callable.rst

```diff
@@ -0,0 +1,107 @@
+    
+    import org.spongepowered.api.Server;
+    import org.spongepowered.api.text.Text;
+    import org.spongepowered.api.text.Texts;
+    import org.spongepowered.api.util.command.CommandCallable;
+    import org.spongepowered.api.util.command.CommandException;
+    import org.spongepowered.api.util.command.CommandResult;
+    import org.spongepowered.api.util.command.CommandSource;
+    
+    import com.google.common.base.Optional;
+    
+    public class MyBroadcastCommand implements CommandCallable {
+    
+        private final Optional<Text> desc = Optional.of((Text) Texts.of("Displays a message to all players"));
+        private final Optional<Text> help = Optional.of((Text) Texts.of("Displays a message to all players. It has no color support!"));
+        private final Text usage = (Text) Texts.of("<message>");
+        
+        private final Server server;
+    
+        public MyBroadcastCommand(Server server) {
+            this.server = server;
+        }
+    
+        public Optional<CommandResult> process(CommandSource source, String arguments) throws CommandException {
+            server.broadcastMessage(Texts.of(arguments));
+            return Optional.of(CommandResult.success());
+        }
+    
+        public boolean testPermission(CommandSource source) {
+            return source.hasPermission("myplugin.broadcast");
+        }
+    
+        public Optional<Text> getShortDescription(CommandSource source) {
+            return desc;
+        }
+    
+        public Optional<Text> getHelp(CommandSource source) {
+            return help;
+        }
+    
+        public Text getUsage(CommandSource source) {
+            return usage;
+        }
+    
+        public List<String> getSuggestions(CommandSource source, String arguments) throws CommandException {
+            return Collections.emptyList();
+        }
+    }
+
+.. tip::
+
+    See the `documentation for CommandCallable <http://spongepowered.github.io/SpongeAPI/org/spongepowered/api/service/command/CommandService.html>`_ for the purposes of each method in this example.
+
+Registering the command
+=======================
+
+Now we can register the class in the ``CommandService``. The ``CommandService`` stands as the manager for watching what commands get typed into chat, and redirecting them to the right command handler.
+To register your command, use the method ``CommandService.register()``, passing your plugin, an instance of the command, and any needed aliases as parameters.
+
+.. code-block:: java
+
+    CommandService cmdService = game.getCommandDispatcher();
+    cmdService.register(plugin, new MyBroadcastCommand(server), "message", "broadcast");
+
+.. note::
+
+    The arguments after the new instance of your command are the aliases to register for the command. You can add as many Strings as you want.
+    The first alias that isn't used by another command becomes the primary alias. This means aliases used by another command are ignored.
```

Comment:
Do you in detail which effects event type and event order have?

How can I override MC commands?